diff --git a/inc/menu.php b/inc/menu.php
index 6e1c301..44cb174 100644
--- a/inc/menu.php
+++ b/inc/menu.php
@@ -20,7 +20,7 @@ function mm_menu_icon() {
 	?>
 	<style type="text/css">
 		li.toplevel_page_mojo-themes .wp-menu-image{
-			background-image: url( '<?php echo MM_ASSETS_URL . "img/mojo-icon-22.png"; ?>' );
+			background-image: url( '<?php echo mm_brand( MM_ASSETS_URL . "img/icon-22-%s.png" ); ?>' );
 			background-position: 7px;
 			background-repeat: no-repeat;
 		}
diff --git a/mojo-marketplace.php b/mojo-marketplace.php
index f078a81..8ae59df 100644
--- a/mojo-marketplace.php
+++ b/mojo-marketplace.php
@@ -34,6 +34,7 @@ require_once( MM_BASE_DIR . 'inc/spam-prevention.php' );
 require_once( MM_BASE_DIR . 'inc/updates.php' );
 require_once( MM_BASE_DIR . 'inc/coming-soon.php' );
 require_once( MM_BASE_DIR . 'inc/tests.php' );
+mm_require( MM_BASE_DIR . 'inc/branding.php' );
 require_once( MM_BASE_DIR . 'inc/editor-prompt.php' );
 mm_require( MM_BASE_DIR . 'inc/sso.php' );
 mm_require( MM_BASE_DIR . 'vendor/jetpack/jetpack-onboarding/jetpack-onboarding.php' );
diff --git a/pages/header.php b/pages/header.php
index 326afb9..d502a24 100644
--- a/pages/header.php
+++ b/pages/header.php
@@ -1,6 +1,6 @@
 <header id="header" class="navbar navbar-default">
-		<div class="header-block bg-cover" style="background-image: url('<?php echo MM_ASSETS_URL; ?>img/header-bg-001.jpg');">
-			<span data-srcset="<?php echo MM_ASSETS_URL; ?>img/header-bg-001.jpg, <?php echo MM_ASSETS_URL; ?>img/header-bg-001-2x.jpg 2x"></span>
+		<div class="header-block bg-cover" style="background-image: url('<?php echo mm_brand( MM_ASSETS_URL . 'img/header-bg-%s.jpg' ); ?>');">
+			<span data-srcset="<?php echo mm_brand( MM_ASSETS_URL . 'img/header-bg-%s.jpg' ); ?>, <?php echo mm_brand( MM_ASSETS_URL . 'img/header-bg-%s-2x.jpg' );?> 2x"></span>
 			<nav>
 				<div class="container">
 					<div class="inner-holder">
@@ -18,6 +18,7 @@
 								<span class="icon-bar hidden-sm hidden-xs"></span>
 								<span class="icon-bar hidden-sm hidden-xs"></span>
 							</button>
+							<!--<span class="powered">Powered By <a href="#"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo mm_brand( MM_ASSETS_URL . 'img/logo-icon-%s.svg' ); ?>"></a></span>-->
 							<div class="dropdown-menu">
 								<div class="avatar-block">
 									<div class="avatar"><a href="#"><i class="icon-avatar"></i></a></div>
@@ -34,7 +35,7 @@
 						</div>
 						<div class="header-holder clearfix">
 							<a class="navbar-brand" href="<?php echo esc_url( add_query_arg( array( 'page' => 'mojo-themes' ), admin_url( 'admin.php' ) ) ); ?>">
-								<img src="<?php echo MM_ASSETS_URL; ?>img/logo-icon.svg" width="250" height="42" alt="MOJO marketplace">
+								<img src="<?php echo mm_brand( MM_ASSETS_URL . 'img/logo-icon-%s.svg' ); ?>" width="250" height="42" alt="Marketplace">
 							</a>
 							<form class="navbar-form form-inline navbar-right text-right" role="search" action="admin.php" method="GET">
 								<div class="form-group">
diff --git a/pages/mojo-graphics.php b/pages/mojo-graphics.php
index fff802a..e3fb2e9 100644
--- a/pages/mojo-graphics.php
+++ b/pages/mojo-graphics.php
@@ -45,7 +45,7 @@ if ( ! is_wp_error( $response ) ) {
 	$items = $api->items;
 
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<div class="container">
 		<?php
@@ -158,6 +158,7 @@ if ( ! is_wp_error( $response ) ) {
 			mm_partner_offers( 'graphics-banner-bottom' );
 			mm_pagination( $api->page, $api->pageCount );
 			?>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
diff --git a/pages/mojo-plugins.php b/pages/mojo-plugins.php
index ad1e683..8a6cbab 100644
--- a/pages/mojo-plugins.php
+++ b/pages/mojo-plugins.php
@@ -32,7 +32,7 @@ if ( ! is_wp_error( $response ) ) {
 	$items = $api->items;
 
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<div class="container">
 		<?php
@@ -145,6 +145,7 @@ if ( ! is_wp_error( $response ) ) {
 			mm_partner_offers( 'plugins-banner-bottom' );
 			mm_pagination( $api->page, $api->pageCount );
 			?>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
diff --git a/pages/mojo-search.php b/pages/mojo-search.php
index 0b55679..f98d476 100644
--- a/pages/mojo-search.php
+++ b/pages/mojo-search.php
@@ -14,7 +14,7 @@ if ( ! is_wp_error( $response ) ) {
 	$items = $body->items;
 	$type = 'search';
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<main id="main">
 		<div class="container">
@@ -94,6 +94,7 @@ if ( ! is_wp_error( $response ) ) {
 						<?php
 					}
 					?>
+					<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 					</div>
 				</div>
 			</div>
diff --git a/pages/mojo-services.php b/pages/mojo-services.php
index 342a5fd..d457655 100644
--- a/pages/mojo-services.php
+++ b/pages/mojo-services.php
@@ -42,7 +42,7 @@ if ( ! is_wp_error( $response ) ) {
 	$items = $api->items;
 
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<div class="container">
 		<?php
@@ -140,6 +140,7 @@ if ( ! is_wp_error( $response ) ) {
 			mm_partner_offers( 'services-banner-bottom' );
 			mm_pagination( $api->page, $api->pageCount );
 			?>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
diff --git a/pages/mojo-themes.php b/pages/mojo-themes.php
index 2c505d0..efc3a1f 100644
--- a/pages/mojo-themes.php
+++ b/pages/mojo-themes.php
@@ -44,7 +44,7 @@ if ( ! is_wp_error( $response ) ) {
 	$items = $api->items;
 
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<div class="container">
 		<?php
@@ -157,6 +157,7 @@ if ( ! is_wp_error( $response ) ) {
 			mm_partner_offers( 'themes-banner-bottom' );
 			mm_pagination( $api->page, $api->pageCount );
 			?>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
diff --git a/pages/single-item.php b/pages/single-item.php
index 408a51b..d052615 100644
--- a/pages/single-item.php
+++ b/pages/single-item.php
@@ -10,7 +10,7 @@ if ( ! is_wp_error( $response ) ) {
 	$item = $body->items[0];
 	$type = $item->type;
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<main id="main">
 		<div class="container">
@@ -148,6 +148,7 @@ if ( ! is_wp_error( $response ) ) {
 					</div>
 				</div>
 			</div>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
diff --git a/tests/loader.php b/tests/loader.php
index 24c38b9..a3e07bf 100644
--- a/tests/loader.php
+++ b/tests/loader.php
@@ -1,21 +1,31 @@
 <?php
 
 function mm_jpo_test( $file ) {
-	return mm_ab_test_file( 'jetpack-onboarding-v1.1.1', $file, 'vendor/jetpack/jetpack-onboarding/jetpack-onboarding.php', 'tests/jetpack-onboarding/jetpack-onboarding.php', 33, DAY_IN_SECONDS * 90 );
+	return mm_ab_test_file( 'jetpack-onboarding-v1.1.1', $file, 'vendor/jetpack/jetpack-onboarding/jetpack-onboarding.php', 'tests/jetpack-onboarding/jetpack-onboarding.php', 25, DAY_IN_SECONDS * 90 );
 }
 add_filter( 'mm_require_file', 'mm_jpo_test' );
 
 function mm_jpo_test_exempt() {
-	mm_ab_test_inclusion( 'jetpack-onboarding-v1.1.1-exempt', md5( 'jetpack-onboarding-v1-exempt' ), 25, DAY_IN_SECONDS * 90 );
+	mm_ab_test_inclusion( 'jetpack-onboarding-v1.1.1-exempt', md5( 'jetpack-onboarding-v1-exempt' ), 33, DAY_IN_SECONDS * 90 );
 }
 add_action( 'init', 'mm_jpo_test_exempt', 9 );
 
 function mm_demo_test( $file ) {
-	return mm_ab_test_file( 'mojo-demo-v1', $file, 'pages/mojo-themes.php', 'tests/mojo-themes.php', 33, DAY_IN_SECONDS * 90 );
+	return mm_ab_test_file( 'mojo-demo-v1', $file, 'pages/mojo-themes.php', 'tests/mojo-themes.php', 25, DAY_IN_SECONDS * 90 );
 }
 add_filter( 'mm_require_file', 'mm_demo_test' );
 
 function mm_demo_test_exempt() {
-	mm_ab_test_inclusion( 'mojo-demo-v1-exempt', md5( 'mojo-demo-v1-exempt' ), 25, DAY_IN_SECONDS * 90 );
+	mm_ab_test_inclusion( 'mojo-demo-v1-exempt', md5( 'mojo-demo-v1-exempt' ), 33, DAY_IN_SECONDS * 90 );
 }
-add_action( 'init', 'mm_demo_test_exempt', 9 );
\ No newline at end of file
+add_action( 'init', 'mm_demo_test_exempt', 9 );
+
+function mm_branding_test_exempt( $file ) {
+	return mm_ab_test_file( 'mojo-branding-exempt', $file, 'inc/branding.php', 'tests/branding.php', 25, DAY_IN_SECONDS * 90 );
+}
+add_filter( 'mm_require_file', 'mm_branding_test_exempt' );
+
+function mm_branding_test() {
+	mm_ab_test_inclusion( 'mojo-branding-' . mm_brand() , md5( 'mojo-branding' . mm_brand() ), 33, DAY_IN_SECONDS * 90 );
+}
+add_action( 'init', 'mm_branding_test', 8 );
diff --git a/tests/mojo-themes.php b/tests/mojo-themes.php
index 67e48d5..5f36c38 100644
--- a/tests/mojo-themes.php
+++ b/tests/mojo-themes.php
@@ -33,7 +33,7 @@ if ( ! is_wp_error( $response ) ) {
 	$api = json_decode( $response['body'] );
 	$items = $api->items;
 ?>
-<div id="mojo-wrapper">
+<div id="mojo-wrapper" class="<?php echo mm_brand( 'mojo-%s-branding' );?>">
 	<?php mm_require( MM_BASE_DIR . 'pages/header.php' ); ?>
 	<div class="container">
 		<?php
@@ -133,6 +133,7 @@ if ( ! is_wp_error( $response ) ) {
 			mm_partner_offers( 'themes-banner-bottom' );
 			mm_pagination( $api->page, $api->pageCount );
 			?>
+			<br style="clear: both"/><span class="alignright powered"><a href="<?php echo mm_build_link( 'https://www.mojomarketplace.com' ); ?>"><img height="16" width="156" alt="Mojo Marketplace" src="<?php echo MM_ASSETS_URL . 'img/logo-dark.svg'; ?>"></a></span>
 		</div>
 	</main>
 </div>
